server/routerlicious/packages/lambdas-driver/src/document-router/documentPartition.ts

```diff
@@ -201,4 +203,43 @@ export class DocumentPartition {
 		this.activityTimeoutTime =
 			activityTime !== undefined ? activityTime : cacluatedActivityTimeout;
 	}
+
+	public pause(offset: number) {
+		if (this.paused) {
+			Lumberjack.info("Doc partition already paused, returning early.", {
+				documentId: this.documentId,
+				tenantId: this.tenantId,
+			});
+			return;
+		}
+		this.paused = true;
+
+		this.q.pause();
+		this.q.remove(() => true); // flush all the messages in the queue since kafka consumer will resume from last successful offset
+
+		if (this.lambda?.pause) {
+			this.lambda.pause(offset);
+		}
+		Lumberjack.info("Doc partition paused", {
+			documentId: this.documentId,
+			tenantId: this.tenantId,
+		});
+	}
+
+	public resume() {
+		if (!this.paused) {
+			Lumberjack.info("Doc partition already resumed, returning early.", {
```

Comment:
nit: I think both this log and the "already paused" log should be `Lumberjack.warn` to help call attention to unexpected cases.